### ISSUE TITLE
How to add Plugins manually (egg)?

### ISSUE THREAD

**Issue by Ltek** [Kabe0/deluge-windscribe#33]
I have tried installing from https://dev.deluge-torrent.org/wiki/Plugins and also, specifically want to install... [ExtractorPlus](https://github.com/d8ahazard/deluge-extractorplus-plugin/releases/tag/1.6.1) manually (Preferences -> Plugins -> 'Install') 

None of them ever install?

**Comment by damsdev1** [comment]
Same here

**Comment by Ltek** [comment]
@Kabe0  any ideas how to fix?

**Comment by Kabe0** [comment]
There is a file under `/config/.config/deluge/core.conf`. It contains any of the plugins you may try to install. You can also specify the plugin inside `/config/.config/deluge/plugins` which is where you can past the raw plugin egg file. This is what I ended up doing to install ItConfig. Once I added it to the plugin folder it was able to be detected by the Deluge container. Make sure to log in and out of Deluge for the plugin to load.

> The /config path will be wherever you bound it on your docker run call for the config volume path

The folder permissions can also be an issue. It technically should be the same as the docker deluge user otherwise it won't have access to modify the plugins. I have noticed that Deluge for some reason sometimes does not properly install plugins in general and garbles up the file extension name in the plugin folder so you might find a 0 KB file in the plugin folder that would probably be the plugin you were trying to install.

See if either of those help.

**Comment by Ltek** [comment]
in [core.conf.txt](https://github.com/Kabe0/deluge-windscribe/files/8920124/core.conf.txt) (my complete file attached to this post) I see 'enabled_plugins' section but nothing that lists all the available plugins?  There are a list of default plugins in the UI that are not in the core.conf and also not in the folder.

```
    "enabled_plugins": [
        "AutoAdd",
        "Extractor",
        "Label"
```

/config is mapped to... /docker/kabe0-deluge so my plugin folder is  `/docker/kabe0-deluge/.config/deluge/plugins`.
I copied these to that folder...
AutoRemovePlus-0.6.1-py2.7.egg
ExtractorPlus-1.6.1-py3.11.egg

I restarted the container (they are still in the folder after restart) but Deluge does not show them in the list of available Plugins in the UI

![Capture](https://user-images.githubusercontent.com/1626802/174102458-f09261e3-47c9-44cf-82f4-c187b1c35c08.JPG)
.

**Comment by Kabe0** [comment]
Ah Ok I think I see it. You are running the wrong python version of the .egg file. The docker container runs Python 3.9.2.
Please download the 3.9 version of the .egg file, past it inside the plugins folder, and it should work as expected

![image](https://user-images.githubusercontent.com/6827834/174112392-6ba80448-2b98-47ef-b9f5-d159a7ef5d75.png)

**Comment by Kabe0** [comment]
@Ltek Any luck?

**Comment by Ltek** [comment]
@Kabe0 yes! you guidance helped. There must be a problem pulling the file from the local PC, via the web UI.

When I use the 'Install' button itself, Deluge is pulling the file from my local PC folder, into the Deluge plugins folder, but its not naming it properly. See screesnshot, I just selected the file named LabelPlus-0.3.2.2-py2.7.egg ...note the file name its putting in the UI and in the folder is not correct. I tried the below work around with LabelPlus-0.3.2.2-py2.7.egg and it did not work :(

If I manually copy over the EGG file into the folder before I select the file from the PC diver (during the Install dialog), it works properly!?

![Capture2](https://user-images.githubusercontent.com/1626802/174313293-2cd7b6ae-1b47-452c-9aa5-852b74ae4736.JPG)
![Capture](https://user-images.githubusercontent.com/1626802/174313295-92d62997-457f-414c-9061-2d3bd96f6470.JPG)

**Comment by Kabe0** [comment]
Yea I saw that happen too with the Deluge image provided by linuxserver. So I think it might be a deluge bug.

**Comment by Ltek** [comment]
> Yea I saw that happen too with the Deluge image provided by linuxserver. So I think it might be a deluge bug.

Seem odd that they would not fix a huge bug like that -- but is seems that Deluge is pretty much a dead product. Maybe its time to look at switching clients?

**Comment by Kabe0** [comment]
Yea that's possible. I have looked at alternatives, though Deluge  has one of the cleanest separations from server to client. A lot of the web UI's provided by other torrent clients don't provide all the actions needed to manage it remotely. Maybe things have changed though as it's been quite a few years since I last looked around. Iptables were such a mess about a year ago I had to stop updating the container till the libraries matured.

**Comment by Kabe0** [comment]
If you have a torrent client suggestion feel free to let me know and I can check I can check it out as a possibility.

**Comment by Ltek** [comment]
Yup, thank you for your help! I did figure out using the UI doesn't work
properly. I copied over the Egg manually to the folder, then used the UI
'install' button... Then it worked. Not sure why but it's easy and quick so
no worries.
